frontend/src/metabase/dashboard/components/DashCard/utils.ts

```diff
@@ -82,36 +82,39 @@ export function getMappingOptionByTarget<T extends DashboardCard>(
   // need to ignore such references here
   const fieldRef = normalizedTarget[1];
   if (isTemplateTagReference(fieldRef)) {
-    return;
+    return undefined;
   }
 
-  const { query, stageIndex, columns } = getParameterColumns(
-    question,
-    parameter,
-  );
+  const { query, columns } = getParameterColumns(question, parameter);
+  const stageIndexes = _.uniq(columns.map(({ stageIndex }) => stageIndex));
```

Comment:
Nit: `_.uniq` can do the map too:
```suggestion
  const stageIndexes = _.uniq(columns, ({ stageIndex }) => stageIndex);
```

Reply:
It's not the same - the original code produces an array of stage indexes, and your suggestion produces an array of columns.